### PR TITLE
Cities api

### PR DESCRIPTION
**Pull request by EugeneJoe**

### CODE REVIEW

File: api/v1/views/cities.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+from models.city import City
+
+
+@app_views.route('/states/<state_id>/cities', methods=['GET'],
+                 strict_slashes=False)
+def get_state_city_objects(state_id):
+    """ Return cities for a given state object """
+    try:
+        state_obj = storage.get(State, state_id)
+        state_cities = [c.to_dict() for c in state_obj.cities()]
```
Comment (Crispy-rw) [inline api/v1/views/cities.py]: state_cities = [c.to_dict() for c in state_obj.cities]

File: api/v1/views/cities.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        return jsonify(state_cities)
+    except:
+        abort(404)
+
+@app_views.route('/cities/<city_id>', methods=['GET'], strict_slashes=False)
+def city_objects(city_id):
+    """ Retrieves the list of all City objects """
+    city_objs = storage.all(City)
```
Comment (Crispy-rw) [inline api/v1/views/cities.py]: if am not mistaken, we need to get one city using the city_id passed on the URL. Maybe u can use `city_objs = storage.get(City, city_id)` and then return its dictionary `return jsonify(city_objs.to_dict())`

File: api/v1/views/cities.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+        content = request.get_json()
+        city = City()
+        city.name = content['name']
+        city.state_id = state_id
+        city.save()
+        return make_response(jsonify(city.to_dict()), 201)
+
+@app_views.route('/cities/<city_id>', methods=['PUT'], strict_slashes=False)
+def update_city_object(state_id):
```
Comment (Crispy-rw) [inline api/v1/views/cities.py]:
```suggestion
def update_city_object(city_id):
```

File: api/v1/views/cities.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    city_obj = storage.get(City, city_id)
+    if city_obj == None:
+        abort(404)
+    if not request.get_json():
+        abort(404, 'Not a JSON')
+    content = request.get_json()
+    nope = ['id', 'created_at', 'updated_at']
+    for key, value in content.items():
+        if key not in nope:
+            setattr(state_obj, key, value)
```
Comment (Crispy-rw) [inline api/v1/views/cities.py]:
```suggestion
            setattr(city_obj, key, value)
```

File: api/v1/views/cities.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    storage.save()
+    return make_reponse(jsonify(state_obj.to_dict()), 200)
```
Comment (Crispy-rw) [inline api/v1/views/cities.py]:
```suggestion
    return make_response(jsonify(state_obj.to_dict()), 200)
```

**Review by Crispy-rw — CHANGES_REQUESTED** [review]

File: api/v1/views/cities.py (hunk 2 above)
Comment (EugeneJoe) [reply api/v1/views/cities.py]: Yes! I don't know what I was doing. Must have been tired.

File: api/v1/views/cities.py (hunk 3 above)
Comment (EugeneJoe) [reply api/v1/views/cities.py]: I must have been really tired. I'll change that

File: api/v1/views/cities.py (hunk 4 above)
Comment (EugeneJoe) [reply api/v1/views/cities.py]: This too

File: api/v1/views/cities.py (hunk 1 above)
Comment (EugeneJoe) [reply api/v1/views/cities.py]: Ah. I was going off of the version 2 project where we had a cities method but that was actually for use with the file storage system, which we aren't using in this one. I'll make the change.

**Review by Crispy-rw — APPROVED** [review]
Good Job

**Pull request merged**